### ISSUE TITLE
Handle when wp-content is symlinked to a folder outside document root

### ISSUE THREAD

**Issue by koekaverna** [rosell-dk/webp-express#315]
I have a site **/srv/site/** and a wordpress in `/srv/wordpress/web`.
Also I have a symlink `web` -> `/srv/wordpress/web`

`index.php` is in `web` folder and `DOCUMENT_ROOT` is `/srv/site`, but images are in **/srv/wordpress/web/app/uploads**. So it fails to check `absPathIsInDocRoot`.

As a solution I suggest to check `WP_CONTENT_DIR` if `DOCUMENT_ROOT` check fails
For classic wordpress `WP_CONTENT_DIR` is `wordpress/wp-content` and upload folder is `wordpress/wp-content/uploads`
For Badrock `WP_CONTENT_DIR` is `web/app` and upload dir is `web/app/uploads`

**Comment by rosell-dk** [comment]
Hi,

I have a test site where I'm using symlinks and that works.
I also have two test sites where I moved Wordpress according to "method 1" and "method 2" as described [here](https://wordpress.org/support/article/giving-wordpress-its-own-directory/)

What is different here seems to be that you moved Wordpress out of document root.

I have been wanting not to rely on DOCUMENT_ROOT for a while as it is not a part of PHP and in some rare cases doesn't work. I have also been wanting to allow configuring which parts of Wordpress that WebP Express works in (ie "Only uploads", "Uploads and active theme", "All Wordpress, including admin"). And this work is actually the next bigger change that I'm going to do, as I need that in place before I can start working on the file manager-like interface for converting, which I'd very much like to do.

**Comment by rosell-dk** [comment]
And doing the above two things will also mean that your use-case will work. I will make sure that will be covered

**Comment by rosell-dk** [comment]
Indeed there is progress :)
I'm very close to releasing 0.15.0. It will probably be Monday. I just need to do some final tests - including testing the use-case described above.

**Comment by rosell-dk** [comment]
You are welcome to beta-test. The "master" branch is currently stable and working

**Comment by rosell-dk** [comment]
Just tested it. The rewrite rules unfortunately does not work when wp-content is symlinked. I'm working on finding a solution

**Comment by rosell-dk** [comment]
This works now in most setups. It however does not work yet when one both symlinks *wp-content* AND sets WP_CONTENT_DIR to the physical path. We need the unresolved path. I have a fix in mind, which requires *unresolving* the physical paths. It is a bit complex and will take some time to implement and test. As this is probably a rare situation, that fix is postponed to 0.15.1 (#335)

**Comment by corbolais** [comment]
Hi @rosell-dk ,
Thank you for the webp express plugin. I just started using it.

For security reasons, support for .htaccess style files is disabled, so after I added the rules manually, it is basically working now.

One thing, though, is nagging me: `DocumentRoot` is set to `/usr/share/wordpress` but `WP_CONTENT_DIR` is set to `/var/lib/wordpress/siteA/`, and this results in:

`[proxy_fcgi:error] AH01071: Got error 'PHP message: Path is outside resolved document root (/usr/share/wordpress). input:/var/lib/wordpress/siteA/favicon.ico`

Is the above related to the this issue #315? Is there already a fix in the pipeline or is it a PEBKAC?

Thank you for your feedback.
